File: packages/js-sdk/src/utils/filesystem.ts
import path from 'path-browserify'

import { Logger } from '../sandbox/sandboxConnection'

export const resolvePath = (
  inputPath: string,
  cwd: string | undefined,
  logger: Logger,
): string => {
  let result: string
  if (inputPath.startsWith('./')) {
    result = path.posix.join(cwd || '/home/user', inputPath)
    if (!cwd) {
      logger.warn?.(
        `Path starts with './' and cwd isn't set. The path '${inputPath}' will evaluate to '${result}', which may not be what you want.`,
      )
    }
    return result
  }

  if (inputPath.startsWith('../')) {
    result = path.posix.join(cwd || '/home/user', inputPath)
    if (!cwd) {
      logger.warn?.(
        `Path starts with '../' and cwd isn't set. The path '${inputPath}' will evaluate to '${result}', which may not be what you want.`,
      )
    }
    return result
  }
  if (inputPath.startsWith('~/')) {
    result = path.posix.join(cwd || '/home/user', inputPath.substring(2))
    if (!cwd) {
      logger.warn?.(
        `Path starts with '~/' and cwd isn't set. The path '${inputPath}' will evaluate to '${result}', which may not be what you want.`,
      )
    }
    return result
  }

  if (!inputPath.startsWith('/') && cwd) {
    return path.posix.join(cwd, inputPath)
  }

  return inputPath
}

File: packages/python-sdk/e2b/utils/filesystem.py
import warnings

from pathlib import PurePosixPath
from typing import Optional


def resolve_path(path: str, cwd: Optional[str] = None) -> str:
    warning = f"Path starts with {{0}} and cwd isn't set. The path {path} will evaluate to `{{1}}`, which may not be what you want."
    if path.startswith("./"):
        result = PurePosixPath(cwd or "/home/user", path).as_posix()
        if not cwd:
            warnings.warn(warning.format("./", result))
        return result

    if path.startswith("../"):
        result = PurePosixPath(cwd or "/home/user", path).as_posix()
        if not cwd:
            warnings.warn(warning.format("../", result))
        return result

    if path.startswith("~/"):
        result = PurePosixPath("/home/user", path[2:]).as_posix()
        warnings.warn(warning.format("~/", result))
        return result

    if not path.startswith("/") and cwd:
        return PurePosixPath(cwd, path).as_posix()

    return path
